moveit_planners/pilz_industrial_motion_planner/src/move_group_sequence_action.cpp

```diff
@@ -47,56 +47,80 @@
 #include <moveit/robot_state/conversions.h>
 #include <moveit/trajectory_processing/trajectory_tools.h>
 #include <moveit/utils/message_checks.h>
+#include <moveit/moveit_cpp/moveit_cpp.h>
 
 #include "pilz_industrial_motion_planner/command_list_manager.h"
 #include "pilz_industrial_motion_planner/trajectory_generation_exceptions.h"
 
 namespace pilz_industrial_motion_planner
 {
+static const rclcpp::Logger LOGGER =
+    rclcpp::get_logger("moveit.pilz_industrial_motion_planner.move_group_sequence_action");
 MoveGroupSequenceAction::MoveGroupSequenceAction() : MoveGroupCapability("SequenceAction")
 {
 }
 
 void MoveGroupSequenceAction::initialize()
 {
   // start the move action server
-  ROS_INFO_STREAM("initialize move group sequence action");
-  move_action_server_ = std::make_unique<actionlib::SimpleActionServer<moveit_msgs::MoveGroupSequenceAction>>(
-      root_node_handle_, "sequence_move_group",
-      boost::bind(&MoveGroupSequenceAction::executeSequenceCallback, this, _1), false);
-  move_action_server_->registerPreemptCallback(boost::bind(&MoveGroupSequenceAction::preemptMoveCallback, this));
-  move_action_server_->start();
+  RCLCPP_INFO_STREAM(LOGGER, "initialize move group sequence action");
+  move_action_server_ = rclcpp_action::create_server<moveit_msgs::action::MoveGroupSequence>(
+      context_->moveit_cpp_->getNode(), "sequence_move_group",
+      [](const rclcpp_action::GoalUUID& /* unused */,
+         std::shared_ptr<const moveit_msgs::action::MoveGroupSequence::Goal> /* unused */) {
+        RCLCPP_DEBUG(LOGGER, "Received action goal");
+        return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+      },
+      [this](const std::shared_ptr<
+             rclcpp_action::ServerGoalHandle<moveit_msgs::action::MoveGroupSequence>> /* unused goal_handle */) {
+        RCLCPP_DEBUG(LOGGER, "Canceling action goal");
+        preemptMoveCallback();
+        return rclcpp_action::CancelResponse::ACCEPT;
+      },
+      [this](
+          const std::shared_ptr<rclcpp_action::ServerGoalHandle<moveit_msgs::action::MoveGroupSequence>> goal_handle) {
+        // Return quickly to avoid blocking executor
+        std::thread(std::bind(&MoveGroupSequenceAction::executeSequenceCallback, this, std::placeholders::_1),
+                    goal_handle)
+            .detach();
```

Comment:
I don't think this's a good solution, why not have a separate callback group for this action?

Reply:
Switched to callbackgroup  with ebf83c3